spyder/plugins/variableexplorer/widgets/dataframeeditor.py

```diff
@@ -515,9 +515,20 @@ def __init__(self, parent, model, header, hscroll, vscroll):
             name='copy',
             parent=self)
         self.horizontalScrollBar().valueChanged.connect(
-                        lambda val: self.load_more_data(val, columns=True))
-        self.verticalScrollBar().valueChanged.connect(
-                        lambda val: self.load_more_data(val, rows=True))
+            self._load_more_columns)
+        self.verticalScrollBar().valueChanged.connect(self._load_more_rows)
+
+    def _load_more_columns(self, value):
+        """Load more columns to display."""
+        # Needed to avoid a NameError while fetching data when closing
+        # See spyder-ide/spyder#12034.
+        self.load_more_data(value, columns=True)

```

Comment:
@dalthviz, please catch `NameError` around this line. I think that's necessary to avoid the error.